include/cvc5/cvc5_parser.h

```diff
@@ -268,6 +268,76 @@ class CVC5_EXPORT InputParser
   std::shared_ptr<Parser> d_parser;
 };
 
+/**
+ * Base class for all Parser exceptions.
+ * If thrown, API objects can still be used
+ */
+class CVC5_EXPORT ParserException : public CVC5ApiException
+{
+ public:
+  // Constructors
+  ParserException();
+
+  ParserException(const std::string& msg);
+
+  ParserException(const char* msg);
+
+  ParserException(const std::string& msg,
+                  const std::string& filename,
+                  unsigned long line,
+                  unsigned long column);
+
+  // Destructor
+  ~ParserException() override;
+
+  void toStream(std::ostream& os) const override;
+
+  std::string getFilename() const;
+
+  /** get the line number of the parsing error */
+  unsigned long getLine() const;
+  /** get the column number of the parsing error */
+  unsigned long getColumn() const;
+
+ protected:
+  std::string d_filename;
+  unsigned long d_line;
+  unsigned long d_column;
+}; /* class ParserException */
+
+/**
+ * An end of file exception.
+ * If thrown, API objects can still be used
+ */
+class ParserEndOfFileException : public ParserException
+{
+ public:
+  // Constructors same as ParserException's
+
+  ParserEndOfFileException();
+
+  ParserEndOfFileException(const std::string& msg);
+
+  ParserEndOfFileException(const char* msg);
+
+  ParserEndOfFileException(const std::string& msg,
+                           const std::string& filename,
+                           unsigned long line,
+                           unsigned long column);
```

Comment:
Document doxygen-style so that this is picked up in the auto-generated documentation